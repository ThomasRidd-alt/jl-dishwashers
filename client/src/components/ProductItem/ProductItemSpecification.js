const ProductItemSpecification = props => {
    const specificationRow = (attribute) => {
        return (
            <div className={'item-page__specification-row'} key={attribute.name}>
                <span className={'item-page__specification-row-name'}>{attribute.name}</span>
                <span className={'item-page__specification-row-value'}>{attribute.value}</span>
            </div>
        )
    }

    return (
        <section className={'item-page__specification'}>
            <h2 className={'item-page__heading'}>Product specification</h2>
            {props.specification.map(attribute => specificationRow(attribute))}
        </section>
    )
};

export default ProductItemSpecification