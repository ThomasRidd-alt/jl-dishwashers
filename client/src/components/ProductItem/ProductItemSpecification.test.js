import React from "react";
import {shallow} from 'enzyme';
import ProductItemSpecification from "./ProductItemSpecification";

describe('ProductItemSpecification', () => {
    const id = 1;
    const props = {
        specification: [
            {name: 'attribute 1', value: 'value 1'},
            {name: 'attribute 2', value: 'value 2'},
            {name: 'attribute 3', value: 'value 3'}
        ]
    }
    const component = shallow(<ProductItemSpecification {...props}/>);

    it('renders properly', () => {
        expect(component).toMatchSnapshot()
    })

    it('renders one row per attribute', () => {
        expect(component.find('.item-page__specification-row').length).toEqual(props.specification.length)
    })
})