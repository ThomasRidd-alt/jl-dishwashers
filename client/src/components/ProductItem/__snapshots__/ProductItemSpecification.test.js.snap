// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductItemSpecification renders properly 1`] = `ShallowWrapper {}`;
